server/src/main/java/org/opensearch/search/SearchService.java

```diff
@@ -379,6 +379,13 @@ public class SearchService extends AbstractLifecycleComponent implements IndexEv
         Setting.Property.Dynamic
     );
 
+    public static final Setting<Boolean> SEARCH_MAX_QUERY_STRING_LENGTH_MONITOR_ONLY = Setting.boolSetting(
+        "search.query.max_query_string_length_monitor_only",
+        false,
+        Setting.Property.NodeScope,
+        Setting.Property.Dynamic
+    );
```

Comment:
I am wondering if this setting should rather be called `search.query.max_query_string_length_mode` and can have more than 2 values:

* `IGNORE` - Behaves same as earlier when the `search.query.max_query_string_length` setting was non existent. No logging/monitoring
* `MONITOR` - Monitors and logs requests without taking any action violating the `search.query.max_query_string_length` setting
* `ENFORCE` - Rejects / cancels requests violating the `search.query.max_query_string_length` setting

I don't know if we really need another mode that does both `MONITOR` and `ENFORCE` by logging and actively rejecting as well!?

Reply:
The reason I like an enum based setting to set the `mode`, is extensibility if we need to have a dual logging & rejecting mode in the future without breaking any backward compatibility. 

Having `mode` related settings as booleans can potentially lead to boolean explosion in long term, so we should be accounting for that as well, and that's my major reasoning of having it as enum even though we have 2 reasonable options as of now.

cc: @cwperks